src/Voter/UriPrefixVoter.php

```diff
@@ -51,6 +76,11 @@ public function matchItem(ItemInterface $item)
 
         $content = $item->getExtra('content');
 
+        $request = $this->request;
+        if (null !== $this->requestStack) {
+            $request = $this->requestStack->getMasterRequest();
```

Comment:
i would do a private getRequest method that looks at stack and falls back to the request from the setter.